src/main/scala/temple/Main.scala

```diff
@@ -1,3 +1,18 @@
 package temple
 
-object Main extends App {}
+import temple.DSL.DSLProcessor
+import temple.utils.FileUtils
+
+object Main extends App {
+  val config = new TempleConfig(args)
+  config.subcommand match {
+    case Some(config.generate) => generate(config)
+    case _                     => config.printHelp()
```

Comment:
Can we change this to None, so it’s clearer that it’s programmer error not user error if a case is forgotten here?

Reply:
And shouldn’t we print a specific error message? Or does scallop do that automatically?

Reply:
Ah good catch with None, will change.

With regards to error message, it seems that scallop will perform `sys.exit(1)` if it encounters an error, for example:

```
sbt:temple> run foo
[info] running temple.Main foo
[scallop] Error: Excess arguments provided: 'foo'

Exception: sbt.TrapExitSecurityException thrown from the UncaughtExceptionHandler in thread "run-main-1"
[error] Nonzero exit code: 1
[error] (Compile / run) Nonzero exit code: 1
```

The case caught by `None` here is when no subcommand is provided at all: should we tell them that? The semantics I went for here mimic that of running `docker` without parameters.

Reply:
That should be fine, as long as `temple -a` errors, and it only does this on a bare input

Reply:
Yep, that's exactly it:
```
sbt:temple> run -a
[info] running temple.Main -a
[scallop] Error: Unknown option 'a'

```